lib/methods/ifwapichecktask.cpp

```diff
@@ -525,7 +465,13 @@ void IfwApiCheckTask::ScriptFunc(const Checkable::Ptr& checkable, const CheckRes
 
 			Defer cancelTimeout ([&timeout]() { timeout->Cancel(); });
 
-			DoIfwNetIo(yc, checkable, cmdLine, cr, psCommand, psHost, expectedSan, psPort, *conn, *req, start);
+			DoIfwNetIo(yc, cr, psCommand, psHost, expectedSan, psPort, *conn, *req);
+
+			cr->SetExecutionEnd(Utility::GetTime());
+
+			// Post the check result processing to the global pool not to block the I/O threads,
+			// which could affect processing important RPC messages and HTTP connections.
+			Utility::QueueAsyncCallback(std::ref(handler));
```

Comment:
I'm not 100% sure what `std::ref()` is doing here exactly. If this would actually capture a reference, wouldn't this become a dangling reference once the coroutine exists?